test/src/main/java/com/navercorp/pinpoint/test/plugin/AbstractPinpointPluginTestSuite.java

```diff
@@ -157,10 +162,33 @@ private String resolveTestClassLocation(Class<?> testClass) {
 
 
     private List<String> getClassPathList(String[] classPathCandidates) {
-        List<String> result = new ArrayList<String>();
-
         ClassLoader cl = getClass().getClassLoader();
+        if (cl.getClass().getName().startsWith(JAVA_9_INTERNAL_CLASSLOADER_PREFIX)) {
+            return getJava9ClassPathList(classPathCandidates);
+        }
+        return getClassPathList(cl, classPathCandidates);
+    }
+
+    private List<String> getJava9ClassPathList(String[] classPathCandidates) {
```

Comment:
The design of classloader dependent code seems to be bad.
If everything is fine, your code looks much better.
Could you replace the old code with yours?


```java
     cl = cl.getParent();
     if (cl == null) {
         break;
      }
```
There may be a problem with the above code, but I'm not sure.
Please check that it works properly in all tests.

Reply:
Thank you for your review @emeroad
I'll change the old code and make sure there are no problems with all the tests.
When this test is completed i will commit again :)